src/core/IncomingRequestManager.cpp

```diff
@@ -34,22 +34,54 @@
 #include "IncomingRequestManager.h"
 #include "ContactsManager.h"
 #include "OutgoingContactRequest.h"
+#include "ContactIDValidator.h"
+#include "utils/Useful.h"
+
+#ifdef PROTOCOL_NEW
+#include "protocol/Connection.h"
+#include "protocol/ContactRequestChannel.h"
+#else
 #include "protocol/ContactRequestServer.h"
-#include <QDebug>
+#endif
 
 IncomingRequestManager::IncomingRequestManager(ContactsManager *c)
     : QObject(c), contacts(c)
 {
     connect(this, SIGNAL(requestAdded(IncomingContactRequest*)), this, SIGNAL(requestsChanged()));
     connect(this, SIGNAL(requestRemoved(IncomingContactRequest*)), this, SIGNAL(requestsChanged()));
+
+#ifdef PROTOCOL_NEW
+    // Attach to any ContactRequestChannel on an incoming connection for this identity
+    connect(contacts->identity, &UserIdentity::incomingConnection, this,
+        [this](Protocol::Connection *connection) {
+            qDebug() << "IncomingRequestManager attaching to connection" << connection;
+            connect(connection, &Protocol::Connection::channelCreated, this,
+                [this](Protocol::Channel *channel) {
+                    Protocol::ContactRequestChannel *req = qobject_cast<Protocol::ContactRequestChannel*>(channel);
+                    qDebug() << "IncomingRequestManager attaching to channel" << channel << req << channel->type() << channel->metaObject()->className();
+                    if (req)
+                        attachRequestChannel(req);
+                }
+            );
+        }
+    );
+#endif
 }
 
 void IncomingRequestManager::loadRequests()
 {
     SettingsObject settings(QStringLiteral("contactRequests"));
 
-    foreach (const QString &host, settings.data().keys())
+    foreach (QString host, settings.data().keys())
     {
+#ifdef PROTOCOL_NEW
+        if (!host.endsWith(QStringLiteral(".onion")))
```

Comment:
True. It actually looks like this should be a QByteArray instead, because it's just `toLatin1`'d below.